conformance/lang/values-types-and-variables/nil-literal.balt

```diff
@@ -0,0 +1,173 @@
+Test-Case: output
+Description: Test nil type descriptor with nil literal '()'.
+Labels: nil-type-descriptor, nil-literal
+
+public function main() {
+    () x = ();
+    io:println(x.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test nil type descriptor with nil literal 'null'.
+Labels: nil-type-descriptor, nil-literal
+
+public function main() {
+    () x = null;
+    io:println(x.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test value and assignability of the nil literal.
+Labels: nil-type-descriptor, nil-literal, equality-expr
+
+public function main() {
+    () x = ();
+    () y = null;
+    io:println(x == y); // @output true
+
+    x = y;
+    io:println(x.toBalString()); // @output ()
+
+    x = ();
+    y = x;
+    io:println(y.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test nil type descriptor with different nil literal representations.
+Labels: nil-type-descriptor, nil-literal
+
+public function main() {
+    () x = ( );
+    io:println(x.toBalString()); // @output ()
+
+    x = (    );
+    io:println(x.toBalString()); // @output ()
+
+    x = (
+    );
+    io:println(x.toBalString()); // @output ()
+
+    x = (());
+    io:println(x.toBalString()); // @output ()
+
+    x = (null);
+    io:println(x.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test nil literal '()' with function call.
+Labels: function-type-descriptor, nil-type-descriptor, nil-literal
+
+public function main() {
+    testNil(());
+}
+
+function testNil(() x) {
+    io:println(x.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test nil literal 'null' with function call.
+Labels: function-type-descriptor, nil-type-descriptor, nil-literal
+
+public function main() {
+    testNil(null);
+}
+
+function testNil(() x) {
+    io:println(x.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test nil literal '()' with optional type descriptor.
+Labels: boolean-type-descriptor, floating-point-type-descriptor, int-type-descriptor, optional-type-descriptor,
+        nil-literal
+
+public function main() {
+    boolean? a = ();
+    io:println(a.toBalString()); // @output ()
+
+    decimal? b = ();
+    io:println(b.toBalString()); // @output ()
+
+    float? c = ();
+    io:println(c.toBalString()); // @output ()
+
+    int? d = ();
+    io:println(d.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test nil literal 'null' with optional type descriptor.
+Labels: boolean-type-descriptor, floating-point-type-descriptor, int-type-descriptor, optional-type-descriptor,
```

Comment:
Didn't we say these would come under optional type-descriptor tests?

Reply:
I think we had decided to add few test cases related to this area.